### PR TITLE
Move pubinternal types in Mvc.Razor and Mvc.RazorPages to internal

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -10,3 +10,5 @@
...
 [assembly: InternalsVisibleTo("Microsoft.AspNetCore.Mvc.TagHelpers.Test, PublicKey=0024000004800000940000000602000000240000525341310004000001000100f33a29044fa9d740c9b3213a93e57c84b472c84e0b8a0e1ae48e67a9f8f6de9d5f7f3d52ac23e48ac51801f1dc950abe901da34d2a9e3baadb141a17c77ef3c565dd5ee5054b91cf63bb3c6ab83f72ab3aafe93d0fc3c2348b764fafb0b1c0733de51459aeab46580384bf9d74c4e28164b7cde247f891ba07891c9d872ad2bb")]
+
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Razor/Properties/AssemblyInfo.cs]: Which tests fail without this and could the failures be avoided by using the `internal` types directly (not Moq-ing them)?

If this is required for functional or integration tests, something is wrong. Our customers might need some `internal` types if so.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Filters/PageHandlerPageFilter.cs [hunk 2]
```diff
@@ -3,9 +3,9 @@
 
 using System;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
 
-namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+namespace Microsoft.AspNetCore.Mvc.Filters
 {
     public class PageHandlerPageFilter : IAsyncPageFilter, IOrderedFilter
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.RazorPages/Filters/PageHandlerPageFilter.cs]: Is this included in our upcoming review of newly `public` types?

File: src/Microsoft.AspNetCore.Mvc.Razor/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (Eilon) [reply src/Microsoft.AspNetCore.Mvc.Razor/Properties/AssemblyInfo.cs]: We do this IVT all the time. I don't have any particular concern about it.

Comment (dougbu) [reply src/Microsoft.AspNetCore.Mvc.Razor/Properties/AssemblyInfo.cs]: I agree but wanted to confirm w.r.t. functional tests in particular.

Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.Razor/Properties/AssemblyInfo.cs]: It complains about `IRazorViewEngineFileProviderAccessor` being internal. We don't want to make the type public, the programming model for changing `IFileProvider` for views is via `RazorViewEngineOptions`.

File: src/Microsoft.AspNetCore.Mvc.RazorPages/Filters/PageHandlerPageFilter.cs (hunk 2 above)
Comment (pranavkm) [reply src/Microsoft.AspNetCore.Mvc.RazorPages/Filters/PageHandlerPageFilter.cs]: Missed updating this.

**Comment by pranavkm** [comment]
> Also, did I miss any newly-public types? I only caught PageHandlerPageFilter.

Fixed that. There aren't any `pubinternal` types made public in this change.

**Pull request merged**